cpp/src/io/parquet/page_enc.cu

```diff
@@ -352,13 +352,19 @@ __global__ void __launch_bounds__(128)
           ? frag_g.num_leaf_values * util::div_rounding_up_unsafe(ck_g.dict_rle_bits, 8)
           : frag_g.fragment_data_size;
       // TODO (dm): this convoluted logic to limit page size needs refactoring
-      size_t this_max_page_size = (values_in_page * 2 >= ck_g.num_values)   ? 256 * 1024
-                                  : (values_in_page * 3 >= ck_g.num_values) ? 384 * 1024
-                                                                            : 512 * 1024;
+      long this_max_page_size = (values_in_page * 2 >= ck_g.num_values)   ? 256 * 1024
```

Comment:
You may also need to modify the line 360 below to avoid comparing between `size_t` and `long`:
```suggestion
if(this_max_page_size > static_cast<long>(max_page_size_bytes)) {
  this_max_page_size = max_page_size_bytes;
}
```

Reply:
The should also be similar issue with comparing `this_max_page_size` with other variables below but I can't figure out an efficient way to fix all.

Reply:
Huh. Doesn't compiler complain about signed-unsigned comparison?

Reply:
I have no idea. Sometimes it is very aggressive. Some other times it just silently ignored.

Reply:
tbh I expected the compiler to yell at me about this...didn't realize `min()` was overloaded so much (unlike `std::min`).  Fixed.

Reply:
Yeah I've been working with cuda code/`nvcc` long enough time and no longer trust in its ability to yell at me for non-critical warnings.

Reply:
I believe that the remaining usage of `this_max_page_size` is at line 369. Can you please check for signed/unsigned comparison there (and fix if necessary)?

Reply:
I think the comparison at line 369 is ok since it's checking whether the sum of two 32-bit unsigneds is greater than a 64-bit signed.  

Ofc that raises the question of whether using 32 bits for page size is sufficient.  But since pages really should be in the kilobytes, it shouldn't be an issue.  Maybe we should just change the `max_page_size_bytes` parameter to be a `size_type` instead?  But that's getting a little out-of-scope maybe.

Reply:
`size_type` is meant to be used for the number of rows (I'm sure there are many places where it's misused). I prefer to use 64-bit types for byte counting, as much as I agree that page size should never be over 2GB :D
Agreed that this would be out-of-scope for this, very focused, PR.

Reply:
I take that back...if the addition overflows, then the test at 369 might never evaluate true 😮 

Maybe there should be a follow-on PR to clean this up.